standard/stream_links.lua

```diff
@@ -93,11 +95,100 @@ function StreamLinks.processStreams(forwardedInputArgs)
 
 				streamValue = StreamLinks.resolve(lookUpPlatform, streamValue)
 			end
+
+			if FeatureFlag.get('new_stream_format') then
+				local key = StreamLinks.StreamKey(platformName):toString()
+				streams[key] = streamValue
+			end
 			streams[platformName] = streamValue
 		end
 	end
 
 	return streams
 end
 
+--- StreamKey Class
+-- Contains the truple that makes up a stream key
+-- [platform, languageCode, index]
+StreamLinks.StreamKey = Class.new(
+	function (self,...)
+		self:new(...)
+	end
+)
+local StreamKey = StreamLinks.StreamKey
+
+function StreamKey:new(tbl, languageCode, index)
+	local platform
+	-- Input is another StreamKey - Make a copy
+	if StreamKey._isStreamKey(tbl) then
+		platform = tbl.platform
+		languageCode = tbl.languageCode
+		index = tbl.index
+	-- Input is another table, assume format is {platform, languageCode, index}
+	elseif type(tbl) == 'table' then
+		platform, languageCode, index = unpack(tbl)
+	-- All three parameters are supplied
+	elseif languageCode and index then
+		platform = tbl
+	elseif type(tbl) == 'string' then
+		local components = mw.text.split(tbl, '_', true)
+		-- Input is in legacy format (eg. twitch2)
+		if #components == 1 then
+			platform, index = self:_fromLegacy(tbl)
+			languageCode = 'en'
+		-- Input is a StreamKey in string format
+		elseif #components == 3 then
+			platform, languageCode, index = unpack(components)
+		end
+	end
+
+	self.platform = platform
+	self.languageCode = languageCode
+	self.index = tonumber(index)
+	self:_isValid()
+	self.languageCode = self.languageCode:lower()
+end
+
+function StreamKey:_fromLegacy(input)
+	for _, platform in pairs(StreamLinks.countdownPlatformNames) do
+		-- The intersection between countdownPlatformNames and streamPlatformLookupNames are not valid platforms.
```

Comment:
technically the **intersection** is always `{}` as in the first table the objects are of the form `17 = 'twitch2'` while in the second table it is `twitch2 = 'twitch'`

Reply:
Reworded to be technically correct as well